src/server/game/Battlefield/Battlefield.cpp

```diff
@@ -396,6 +396,13 @@ void Battlefield::AskToLeaveQueue(Player* player)
     m_PlayersInQueue[player->GetTeamId()].erase(player->GetGUID());
 }
 
+// Called in WorldSession::HandleHearthAndResurrect
+void Battlefield::PlayerAskToLeave(Player* player)
+{
+    // Player leaving Wintergrasp, teleport to Dalaran.
+    player->TeleportTo(571, 5804.1499f, 624.7710f, 647.7670f, 1.6400f);
```

Comment:
According to https://wow.gamepedia.com/Wintergrasp the players leaving Wintergrasp should instantly use their Hearthstone even if on cooldown. So I suggest the following to trigger the "Hearthstone" spell instead of just teleporting to Dalaran:
```suggestion
    player->CastSpell(player, 8690, true);
```

Reply:
Yeah, looks fine for me. Thank you